pkg/server/ca/manger_health.go

```diff
@@ -0,0 +1,50 @@
+package ca
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/spiffe/spire/pkg/common/health"
+)
+
+// TODO: What would be a good threshold number?
+const failedRotationThreshold = 10
+
+type managerHealth struct {
+	m *Manager
+}
+
+func (h *managerHealth) CheckHealth() health.State {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	err := h.m.rotate(ctx)
```

Comment:
I don't think we should actively call rotate here. When this function returns nil, it doesn't really mean rotation was successful, it could also mean rotation wasn't needed. When rotation IS failing, this call could cause more frequent rotation attempts. We should let the manager decide how often it is attempting rotation.

I think for now we should just rely on the same metric for readiness?

Reply:
Done!